cpp/dolfinx/fem/DirichletBC.h

```diff
@@ -300,12 +314,11 @@ class DirichletBC
   /// @note The size of of `g` must be equal to the block size if `V`.
   /// Use the Function version if this is not the case, e.g. for some
   /// mixed spaces.
-  template <typename S, typename X,
-            typename
-            = std::enable_if_t<std::is_convertible_v<S, T>
-                               or std::is_convertible_v<S, std::span<const T>>>>
+  template <typename S, typename X>
     requires std::is_convertible_v<std::remove_cvref_t<X>,
                                    std::vector<std::int32_t>>
+             && (std::is_convertible_v<S, T>
+                 || std::is_convertible_v<S, std::span<const T>>)
```

Comment:
It would be easier to read if the order of the `requires` followed the same order as the function arguments.